### PR TITLE
Or converted to add, added tests for negative value of scale

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/machinelearning/issues/1802

### CODE REVIEW

**Comment by eerhardt** [comment]
Please enable these tests as well since you are fixing this bug:

https://github.com/dotnet/machinelearning/blob/f0fb2a0cc5316dc5abce55e3f0893e4d3d94ba1e/test/Microsoft.ML.Core.Tests/UnitTests/TestVBuffer.cs#L42-L64

File: src/Microsoft.ML.CpuMath/AvxIntrinsics.cs [hunk 1]
```diff
@@ -934,7 +934,7 @@ public static unsafe void Scale(float scale, Span<float> dst)
                     result = Avx.And(result, leadingMask);
 
                     temp = Avx.Multiply(scaleVector256, temp);
-                    temp = Avx.Or(temp, result);
+                    temp = Avx.Add(temp, result);
```
Comment (eerhardt) [inline src/Microsoft.ML.CpuMath/AvxIntrinsics.cs]: Instead of using `Add`, which is a little confusing to a new reader, would it make more sense to do the "trailing mask" after the `Avx.Multiply`? That way you can still use `Avx.Or`, which is the more reasonable/obvious thing to do here.

No matter which way we end up, I believe a comment is justified here stating why we are doing what we are doing.
